climada/hazard/low_flow.py

```diff
@@ -579,31 +726,109 @@ def _read_single_nc(filename, yearrange, bbox):
     return data
 
 
+def _xarray_reduce(data, fun=None, percentile=None):
+    """wrapper function to reduce xarray along time axis
+
+    Parameters:
+        data (xarray)
+
+    Optional Parameters:
+        fun (str): function to be applied, either "mean" or "percentile"
+        percentile (num): percentile to be extracted, e.g. 5 for 5th percentile
+            (only if fun=='percentile')
+
+    Returns:
+        xarray
+    """
+    if fun == 'mean':
+        return data.mean(dim='time')
+    if fun[0] == 'p':
+        return data.reduce(np.nanpercentile, dim='time', q=percentile)
+    return None
+
+def _split_bbox(bbox, width=BBOX_WIDTH):
+    """split bounding box into squares, return new set of bounding boxes
+
+    Parameters:
+        bbox (list): geographical bounding box in the form:
+            [lon_min, lat_min, lon_max, lat_max]
+
+    Optional Parameters:
+        width (float): width and height of geographical bounding boxes for loop in degree lat/lon.
+        i.e. the bounding box is split into square boxes with maximum size BBOX_WIDTH*BBOX_WIDTH
+
+    Returns:
+        bbox_list (list): list of bounding boxes of the same format as bbox
+    """
+    if not bbox:
+        bbox = [-180, -85, 180, 85]
+    lons = [bbox[0]] + \
+        [int(idc) for idc in np.arange(np.ceil(bbox[0]+width-1),
+                                       np.floor(bbox[2]-width+1), width)] + [bbox[2]]
+    lats = [bbox[1]] + \
+        [int(idc) for idc in np.arange(np.ceil(bbox[1]+width-1),
+                                       np.floor(bbox[3]-width+1), width)] + [bbox[3]]
+    bbox_list = list()
+    for ilon, _ in enumerate(lons[:-1]):
+        for ilat, _ in enumerate(lats[:-1]):
+            bbox_list.append([lons[ilon], lats[ilat], lons[ilon+1], lats[ilat+1]])
+    return bbox_list
```

Comment:
it's alright to get the 4 values as one argument, but within the method, they should have names.
`lon_min, lat_min, lon_max, lat_max = bbox`